File: WEEKLYCODE#3/migratory_birds/Python/migratory_birds.py
import sys
#taking inputs
n = int(input().strip())
types = list(map(int, input().strip().split(' ')))
counts = [0,0,0,0,0,0]
#using indexed based increment for frequency 
for bird in types:
    counts[bird] += 1
result = 0
#checking for maximum frequency
for x in range(6):
    if counts[x] > counts[result]:
        result = x
#printing result
print(result)

File: WEEKLYCODE#3/migratory_birds/C++/migratory_birds.cpp
#include <iostream>
using namespace std;

int main() {
	int n,i,max,m,ind;
    //taking inputs
    scanf("%d",&n);
    int ar[n];
    for(i=0;i<n;i++)
        scanf("%d",&ar[i]);
    //finding the max element
    max=ar[0];
    for(i=1;i<n;i++)
        if(ar[i]>max)
            max=ar[i];
    max+=1;
    //declaring and assigning an array of size max equal to 0
    int br[max];
    for(i=0;i<max;i++)
        br[i]=0;
    //using index based increment for frequency of a number
    for(i=0;i<n;i++)
        br[ar[i]]+=1;
    m=br[0];
    ind=0;
    //choosing the index with max frequency
    for(i=1;i<max;i++)
        if(m<br[i]){
            m=br[i];
            ind=i;
        }
    //printing result
    printf("%d",ind);
    
	return 0;
}